src/resmom/linux/alps.c

```diff
@@ -4470,7 +5180,41 @@ alps_create_reserve_request(job *pjob, basil_request_reserve_t **req)
 		sizeof(basil_req->batch_id)-1);
 	basil_req->batch_id[sizeof(basil_req->batch_id)-1] = '\0';
 
-	for (i=0; i<num; i++) {
+	/* check for pstate or pgov */
+	for (pres = (resource *)
+				GET_NEXT(pjob->ji_wattr[(int)JOB_ATR_resource].at_val.at_list);
+			pres != NULL;
+			pres = (resource *)GET_NEXT(pres->rs_link)) {
+		char	*pname;
+
+		if ((pstate > 0) && (pgov != NULL))
+			break;
+
+		if (pres->rs_defin == NULL)
+			continue;
+		pname = pres->rs_defin->rs_name;
+		if (pname == NULL)
+			continue;
+
+		if (strcmp(pname, "pstate") == 0) {
+			pstate = atol(pres->rs_value.at_val.at_str);
+			if (pstate <= 0) {
+				snprintf(log_buffer, sizeof(log_buffer),
+					"pstate value \"%s\" could not be used for the reservation",
+					pres->rs_value.at_val.at_str);
+				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB,
+					LOG_DEBUG, pjob->ji_qs.ji_jobid, log_buffer);
+				pstate = 0;
+			}
+			continue;
+		}
+		if (strcmp(pname, "pgov") == 0) {
+			pgov = pres->rs_value.at_val.at_str;
+			continue;
+		}
+	}
```

Comment:
This code seems outside the scope of using the new BASIL 1.3/1.4 interface to fill in information for resources PBS already has.  This seems to be introducing/using new features.